src/main/webapp/WEB-INF/pages/navbar.jsp

```diff
@@ -65,7 +62,17 @@
                 <li class="nav-item">
                     <a class="nav-link" href="${contextPath}/Profile">Profile</a>
                 </li>
-            </c:if>
+            </c:if> 
+            <li class="nav-item">
+                <c:choose>
+                        <c:when test="${pageContext.request.getRemoteUser() == null}"> 
```

Comment:
Does this check return null anytime an user is not logged in? Doesn't have any other corner cases?
I think the logic there should be to show logout when user is logged in. Any other situation, the login should be displayed. Are you sure this type of check completely covers the logic?

Reply:
java.lang.String getRemoteUser(): Returns the login of the user making this request, if the user has been authenticated, or null if the user has not been authenticated. 

https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletRequest.html#getUserPrincipal%28%29